src/main/java/sirius/biz/tenants/deletion/DeleteTenantJob.java

```diff
@@ -0,0 +1,116 @@
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.tenants.deletion;
+
+import sirius.biz.jobs.JobCategory;
+import sirius.biz.jobs.JobFactory;
+import sirius.biz.jobs.batch.BatchJob;
+import sirius.biz.jobs.batch.DefaultBatchProcessFactory;
+import sirius.biz.jobs.params.BooleanParameter;
+import sirius.biz.jobs.params.Parameter;
+import sirius.biz.jobs.params.TenantParameter;
+import sirius.biz.process.ProcessContext;
+import sirius.biz.process.logs.ProcessLog;
+import sirius.biz.tenants.Tenant;
+import sirius.biz.tenants.TenantController;
+import sirius.biz.tenants.Tenants;
+import sirius.biz.tenants.UserAccount;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.PriorityParts;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.nls.NLS;
+import sirius.web.security.Permission;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Deletes a given tenant and the related data.
+ * <p>
+ * Implement and register a {@link TenantDeletionWorkItem} object to implement logic for deleting external tenant data,
+ * like {@link UserAccount UserAccounts} for example.
+ */
+public class DeleteTenantJob extends BatchJob {
+
+    @Part
+    private static Tenants<?, ?, ?> tenants;
+
+    @PriorityParts(TenantDeletionWorkItem.class)
+    private static List<TenantDeletionWorkItem> deletionWorkItems;
+
+    private final Tenant<?> tenant;
+
+    private DeleteTenantJob(ProcessContext process, Tenant<?> tenant) {
+        super(process);
+        this.tenant = tenant;
+    }
+
+    @Override
+    public void execute() throws Exception {
+        boolean simulate = process.getParameter(DeleteTenantJobFactory.SIMULATION_PARAMETER).orElse(true);
+
+        if (simulate) {
+            process.log(ProcessLog.info().withNLSKey("DeleteTenantJob.simulate"));
+        }
+
+        for (TenantDeletionWorkItem deletionWorkItem : deletionWorkItems) {
+            process.log(deletionWorkItem.execute(tenant, simulate));
+        }
+    }
+
+    @Register(classes = JobFactory.class, framework = Tenants.FRAMEWORK_TENANTS)
+    @Permission(TenantController.PERMISSION_MANAGE_TENANTS)
+    public static class DeleteTenantJobFactory extends DefaultBatchProcessFactory {
+
+        protected static final TenantParameter TENANT_PARAMETER =
+                new TenantParameter("tenant", "$Model.tenant").markRequired();
+
+        protected static final BooleanParameter SIMULATION_PARAMETER =
+                new BooleanParameter("simulate", "$DeleteTenantJobFactory.simulate");
+
+        @Part
+        protected Tenants<?, ?, ?> tenants;
+
+        @Override
+        protected String createProcessTitle(Map<String, String> context) {
+            return NLS.fmtr("DeleteTenantJobFactory.title")
+                      .set("name", tenants.fetchCachedTenantName(context.get("tenant")))
```

Comment:
use the parameter to resolve the id into a tenant - this assums the implementation of the parameter which isn't guaranteed